src/masternodes/mn_checks.cpp

```diff
@@ -3083,6 +3097,93 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return mnview.AddBalance(obj.to, obj.amount);
     }
 
+    Res operator()(const CPaybackWithCollateralMessage& obj) const {
+        auto res = CheckCustomTx();
+        if (!res)
+            return res;
+
+        const auto attributes = mnview.GetAttributes();
+        if (!attributes)
+            return Res::Err("Attributes unavailable");
+
+        auto dUsdToken = mnview.GetToken("DUSD");
+        if (!dUsdToken)
+            return Res::Err("Cannot find token DUSD");
+
+        CDataStructureV0 activeKey{AttributeTypes::Token, dUsdToken->first.v, TokenKeys::LoanPaybackCollateral};
+        if (!attributes->GetValue(activeKey, false))
+            return Res::Err("Payback of DUSD loan with collateral is not currently active");
+
+        // vault exists
+        auto vault = mnview.GetVault(obj.vaultId);
+        if (!vault)
+            return Res::Err("Vault <%s> not found", obj.vaultId.GetHex());
+
+        // vault under liquidation
+        if (vault->isUnderLiquidation)
+            return Res::Err("Cannot payback vault with collateral while vault's under liquidation");
+
+        // owner auth
+        if (!HasAuth(vault->ownerAddress))
+            return Res::Err("tx must have at least one input from token owner");
+
+        if (!IsVaultPriceValid(mnview, obj.vaultId, height))
+            return Res::Err("Cannot payback vault with collateral while any of the asset's price is invalid");
+
+        const auto collateralAmounts = mnview.GetVaultCollaterals(obj.vaultId);
+        if (!collateralAmounts) {
+            return Res::Err("Vault has no collaterals");
+        }
+        if (!collateralAmounts->balances.count(dUsdToken->first)) {
+            return Res::Err("Vault does not have any DUSD collaterals");
+        }
+        const auto& collateralDUSD = collateralAmounts->balances.at(dUsdToken->first);
+
+        const auto loanAmounts = mnview.GetLoanTokens(obj.vaultId);
+        if (!loanAmounts) {
+            return Res::Err("Vault has no loans");
+        }
+        if (!loanAmounts->balances.count(dUsdToken->first)) {
+            return Res::Err("Vault does not have any DUSD loans");
+        }
+        const auto& loanDUSD = loanAmounts->balances.at(dUsdToken->first);
+
+        auto rate = mnview.GetInterestRate(obj.vaultId, dUsdToken->first, height);
+        if (!rate)
+            return Res::Err("Cannot get interest rate for this token (DUSD)!");
+        auto subInterest = TotalInterest(*rate, height);
+
+        // Edge case where interest is greater than collateral
+        if (subInterest > collateralDUSD) {
+            res = mnview.SubVaultCollateral(obj.vaultId, {dUsdToken->first, collateralDUSD});
+            if (!res)
+                return res;
+
```

Comment:
missing the sub from minted amount. subCollateralAmount is taken out of circulation -> needs to reduce the total minted amount.

Reply:
Fixed in https://github.com/DeFiCh/ain/pull/1451/commits/99161a1ce7889ace1d42f260d4fc90c64774baff